src/solver/gemm_bwd.cpp

```diff
@@ -0,0 +1,761 @@
+                                     const conv::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_GEMM
+    auto& handle       = context.GetStream();
+    const auto& conv   = problem.GetConv();
+    const auto& dyDesc = problem.GetIn();
+    const auto& wDesc  = problem.GetWeights();
+
+    const auto spatial_dim = conv.GetSpatialDimension();
+
+    const auto wei_spatial = boost::adaptors::slice(wDesc.GetLengths(), 2, 2 + spatial_dim);
+    const auto out_spatial = boost::adaptors::slice(dyDesc.GetLengths(), 2, 2 + spatial_dim);
+
+    const auto wei_c = wDesc.GetLengths()[1];
+
+    const auto gemm_size = wei_c * std::accumulate(wei_spatial.begin(),
+                                                   wei_spatial.end(),
+                                                   std::size_t(1),
+                                                   std::multiplies<std::size_t>()) *
+                           std::accumulate(out_spatial.begin(),
+                                           out_spatial.end(),
+                                           std::size_t(1),
+                                           std::multiplies<std::size_t>()) *
+                           GetTypeSize(dyDesc.GetType()) * conv.group_count;
+
+    if(gemm_size > MAX_MEM_ALLOC_SZ)
+        return 0;
+    return gemm_size;
+#else
+    std::ignore = context;
+    std::ignore = problem;
+
+    return 0;
+#endif
+}
+
+bool GemmBwdRest::IsApplicable(const ExecutionContext& context,
+                               const conv::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_GEMM
+    if(!GemmBwdBase::IsApplicable(context, problem))
+        return false;
+
+    return !GemmBwd1x1_stride2{}.IsApplicable(context, problem) &&
+           !GemmBwd1x1_stride1{}.IsApplicable(context, problem);
+#else
+    std::ignore = context;
+    std::ignore = problem;
+    return false;
+#endif
+}
+
+ConvSolution GemmBwdRest::GetSolution(const ExecutionContext& context,
+                                      const conv::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_GEMM
+    const auto& dyDesc = problem.GetIn();
+    const auto& wDesc  = problem.GetWeights();
+    const auto& dxDesc = problem.GetOut();
+    const auto& conv   = problem.GetConv();
+
+    const auto group_count = conv.group_count;
+
+    const auto spatial_dim  = conv.GetSpatialDimension();
+    const auto in_spatial_  = boost::adaptors::slice(dxDesc.GetLengths(), 2, 2 + spatial_dim);
+    const auto wei_spatial_ = boost::adaptors::slice(wDesc.GetLengths(), 2, 2 + spatial_dim);
+    const auto out_spatial_ = boost::adaptors::slice(dyDesc.GetLengths(), 2, 2 + spatial_dim);
+    const auto in_spatial   = std::vector<std::size_t>(in_spatial_.begin(), in_spatial_.end());
+    const auto wei_spatial  = std::vector<std::size_t>(wei_spatial_.begin(), wei_spatial_.end());
+    const auto out_spatial  = std::vector<std::size_t>(out_spatial_.begin(), out_spatial_.end());
```

Comment:
@asroy @fronteer @shurale-nkn Notice this. We pass everything to InvokerFactory (and to Invoker) by value. The copies of values are captured in Factory (and Invoker) lambda instances, so later we can call Factory (or Invoker) from any context.

However `slice` returns _reference_ to the original object. Even if you pass it to an InvokerFactory (or to Invoker) by-value, only reference will be captured . Subsequent invocation of Factory (or Invoker) will cause referencing of non-existing object.

That is why @DrizztDoUrden creates copies passes that copies to the Factory.

Reply:
The alternative is calculating slices inside of invoker from invoke ctx tensor descriptors.

Reply:
Essentially, you would want to capture anything tiny or time-consuming to calculate, leaving only the rest to be calculated in the invoker.